network/src/protocols/network/mod.rs

```diff
@@ -14,24 +14,30 @@ use crate::{
     transport::ConnectionMetadata,
     ProtocolId,
 };
+use aptos_bounded_executor::BoundedExecutor;
 use aptos_channels::aptos_channel;
 use aptos_logger::prelude::*;
 use aptos_short_hex_str::AsShortHexStr;
 use aptos_types::{network_address::NetworkAddress, PeerId};
 use bytes::Bytes;
 use futures::{
-    channel::oneshot,
+    channel::{mpsc, oneshot},
     future,
-    stream::{FilterMap, FusedStream, Map, Select, Stream, StreamExt},
+    stream::{FusedStream, Map, Select, Stream, StreamExt},
     task::{Context, Poll},
 };
+use futures_util::SinkExt;
 use pin_project::pin_project;
 use serde::{de::DeserializeOwned, Serialize};
 use std::{cmp::min, fmt::Debug, marker::PhantomData, pin::Pin, time::Duration};
+use tokio::runtime::Handle;
 
 pub trait Message: DeserializeOwned + Serialize {}
 impl<T: DeserializeOwned + Serialize> Message for T {}
 
+// TODO: do we want to make this configurable?
+const MAX_DESERIALIZATION_QUEUE_SIZE: usize = 1024;
```

Comment:
Why does this need to be bounded? What happens if this queue gets full?

Reply:
The effective current depth of queue here is zero? Trying to read a deserialized message causes trying to read from the upstream channel, do the deserialize, and return the result.

Reply:
I've changed this so that we: (i) provide a queue per peer of max depth 30 (which should be more than enough); and (ii) if the queue fills up (e.g., the application layers fails to handle the deserialized messages fast enough), we'll start dropping messages instead of blocking or consuming large amounts of memory. This seems like the safer approach. Thoughts?